web/src/app/core/services/authentication.service.ts

```diff
@@ -84,11 +84,11 @@ export class AuthenticationService {
 
     // This function used to login via github
     public login(): void {
-        this.spinnerService.setProgressBar(true);
         const provider: auth.GithubAuthProvider = new auth.GithubAuthProvider();
         provider.addScope('repo,admin:repo_hook');
         from(this.afAuth.auth.signInWithPopup(provider))
             .pipe(
+                tap(() => this.spinnerService.setProgressBar(true)),
```

Comment:
Initially we were starting spinner , when we click on login button now after adding tap , its starting spinner once you enter sign in details in popup . Do you want this change intentionally ?

Reply:
My concern is, if someone closes the small popup window without signing in, the progress spinner keeps going and never ends (plus I think it is a bit cleaner code to have it in the `pipe`). So I think it makes more sense for the progress bar to start once the small popup closes after the user has signed in.

Reply:
Thanks for clarification . Looks good 👍